openapi.yaml

```
@@ -0,0 +1,2411 @@
+          oneOf:
+            - $ref: '#/components/schemas/ScheduleInterval'
+          readOnly: true
+        tags:
+          type: array
+          items:
+            $ref: '#/components/schemas/Tag'
+          readOnly: true
+
+    DAGCollection:
+      type: object
+      properties:
+        dag_model:
+          type: array
+          items:
+            $ref: '#/components/schemas/DAG'
+
+    DAGRun:
+      type: object
+      properties:
+        dag_id:
+          type: string
+          readOnly: true
+        execution_date:
+          description: >
+            The execution date. This is the time when the DAG run should be started according
+            to the DAG definition.
+
+            If the specified execution is in use, the creation request fails with an ALREADY_EXISTS error.
+
+            This together with DAG_ID are a unique key.
+          type: string
+          format: date-time
+          readOnly: true
+        start_date:
+          type: string
+          format: date-time
+          description: >
+            The start time. The time when DAG Run was actually created..
+          readOnly: True
+        end_date:
+          type: string
+          format: date-time
+          readOnly: True
+        state:
+          $ref: '#/components/schemas/DagState'
+        run_id:
+          type: string
+          description: >
+            Run ID.
+
+            If not provided, a value will be generated based on execution_date.
+
+            If the specified execution is in use, the creation request fails with an ALREADY_EXISTS error.
+
+            This together with DAG_ID are a unique key.
+        external_trigger:
+          type: boolean
+          default: true
+          readOnly: True
+        conf:
+          type: string
+          description: >
+            JSON object describing additional configuration parameters.
+      required:
+        - dag_id
+
+    DAGRunCollection:
+      type: object
+      properties:
+        dag_runs:
+          type: array
+          items:
+            $ref: '#/components/schemas/DAGRun'
+
+    EventLog:
+      type: object
+      properties:
+        event_log_id:
+          type: integer
+          readOnly: true
+        dttm:
+          type: string
+          readOnly: true
+        dag_id:
+          type: string
+          readOnly: true
+        task_id:
+          type: string
+          readOnly: true
+        event:
+          type: string
+          readOnly: true
+        execution_date:
+          type: string
+          readOnly: true
+        owner:
+          type: string
+          readOnly: true
+        extra:
+          type: string
+          readOnly: true
+
+    EventLogCollection:
+      type: object
+      properties:
+        event_logs:
+          type: array
+          items:
+            $ref: '#/components/schemas/EventLogCollection'
+
+    ImportError:
+      type: object
+      properties:
+        import_error_id:
```

Can be simplified to `error_id` or `id` I guess

```suggestion
        error_id:
```

---

I try to keep some conventions. If the object is called Import Error, the field with the identifier is `import_error_id`. The `id` field is too generic and can easily be confused. I would also like the object ID to have consequences throughout the API. If we use the name dag_id in one object, then `dag_id` must be used in every place, even if in the given situation the `id` will clearly describe the field.